tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/BlockingMiddleware.cs

```diff
@@ -0,0 +1,116 @@
+﻿// <copyright file="BlockingMiddleware.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+#if !NETFRAMEWORK
+using System.Threading.Tasks;
+using Datadog.Trace.AppSec;
+using Datadog.Trace.AppSec.Transports;
+using Microsoft.AspNetCore.Http;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore;
+
+internal class BlockingMiddleware
+{
+    private readonly bool _runSecurity;
+    private readonly bool _endPipeline;
+    private readonly RequestDelegate? _next;
+
+    internal BlockingMiddleware(RequestDelegate? next = null, bool runSecurity = false, bool endPipeline = false)
+    {
+        _next = next;
+        _runSecurity = runSecurity;
+        _endPipeline = endPipeline;
+    }
+
+    private static Task WriteResponse(HttpContext context, SecuritySettings settings, out bool endedResponse)
+    {
+        var httpResponse = context.Response;
+        if (!httpResponse.HasStarted)
+        {
+            httpResponse.Clear();
+            foreach (var cookie in context.Request.Cookies)
+            {
+                httpResponse.Cookies.Delete(cookie.Key);
+            }
+
+            httpResponse.Headers.Clear();
+            httpResponse.StatusCode = 403;
+            var template = settings.BlockedJsonTemplate;
+            if (context.Request.Headers["Accept"].ToString().Contains("text/html"))
+            {
+                httpResponse.ContentType = "text/html";
+                template = settings.BlockedHtmlTemplate;
+            }
+            else
+            {
+                httpResponse.ContentType = "application/json";
+            }
+
+            endedResponse = true;
+            return httpResponse.WriteAsync(template);
+        }
+
+        context.Response.Body.Dispose();
```

Comment:
Hmm, so in this case, the response _has_ already started, but you're calling `Dispose()` - I think that could cause the app to start writing to a disposed stream, which may throw `ObjectDisposed` exceptions? 🤔 Maybe it would be better to abort the connection entirely with context.Abort()? I'm not sure..